pkg/certificate/cleanup.go

```diff
@@ -72,14 +76,35 @@ func (m *Manager) cleanUpCABundle() error {
 	return nil
 }
 
-func (m *Manager) cleanUpExpiredCertificates(certificates []*x509.Certificate) []*x509.Certificate {
+func (m *Manager) cleanUpCertificates(certificates []*x509.Certificate) []*x509.Certificate {
+	logger := m.log.WithName("cleanUpCertificates")
+	logger.Info("Cleaning up expired or beyond overlap duration limit at CA bundle")
+	// There is no overlap
+	if len(certificates) <= 1 {
+		return certificates
+	}
+
 	now := m.now()
 	// create a zero-length slice with the same underlying array
 	cleanedUpCertificates := certificates[:0]
-	for _, certificate := range certificates {
-		if certificate.NotAfter.After(now) {
-			cleanedUpCertificates = append(cleanedUpCertificates, certificate)
+	for i, certificate := range certificates {
+		logger.Info("Checking certificate for cleanup", "now", now, "caOverlapDuration", m.caOverlapDuration, "NotBefore", certificate.NotBefore, "NotAfter", certificate.NotAfter)
+
+		// Expired certificate are cleaned up
+		caExpirationDate := certificate.NotAfter
+		if now.After(caExpirationDate) {
+			logger.Info("Cleaning up expired certificate", "now", now, "NotBefore", certificate.NotBefore, "NotAfter", certificate.NotAfter)
+			continue
+		}
+
+		// Certificates that are not the last appended one (the last generated from a rotation) and is pass CA Overlap Duration limit
```

Comment:
please rephrase comment (suggestion below): 
`Clean up certificates that pass CA Overlap Duration limit, except for the last appended one (i.e. the last generated from a rotation) since we need at least one valid certificate`